File: backend/app/schemas/tender_department.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class TenderDepartmentCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)


class TenderDepartmentUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    is_active: bool | None = None


class TenderDepartmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    is_active: bool
    created_at: datetime

File: frontend/src/types/tender-department.ts
export interface TenderDepartment {
  id: string;
  name: string;
  is_active: boolean;
  created_at: string;
}

export interface TenderDepartmentCreate {
  name: string;
}

export interface TenderDepartmentUpdate {
  name?: string;
  is_active?: boolean;
}
